### PR TITLE
AMX RTB: remove use of auctionId

### PR DESCRIPTION
**Pull request by nickjacob**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [X] Other

## Description of change

Remove use of bidderRequest.auctionId, replacing it with ortb2 "tid" / transaction ID. For compliance with: https://github.com/prebid/Prebid.js/issues/9781#event-9417751649

### CODE REVIEW

File: modules/amxBidAdapter.js [hunk 1]
```diff
@@ -302,8 +303,13 @@ export const spec = {
...
         };
 
+    const transactionId = deepAccess(
+      bidderRequest,
+      'ortb2.ext.tid',
+      deepAccess(bidRequests[0], 'ortb2Imp.ext.tid', generateUUID()));
+
     const payload = {
-      a: bidderRequest.auctionId,
+      a: transactionId,
```
Comment (patmmccann) [inline modules/amxBidAdapter.js:312]: Quick question @nickjacob ; do you plan to put this field in ortb2 bid requests to downstream parties ion either source.tid or imp.ext.tid? If so, imp.ext.tid should not fall back to a value that isn't shared by all bidders. The openrtb spec doesnt' allow for exchanges to make up their own value here. Perhaps you would want to pass yourself an indicator of if this value is shared or null if you can't take the null here?

File: modules/amxBidAdapter.js [hunk 2]
```diff
@@ -302,8 +303,13 @@ export const spec = {
           bidRequestsCount: 0,
         };
 
+    const transactionId = deepAccess(
```
Comment (dgirardi) [inline modules/amxBidAdapter.js:306]: What are you using this for? I can't make sense of this if `payload.a` is supposed to have any meaning at all when it gets to your server:

 - `auctionid` and `transactionId` are not meant to be mixed, picking them interchangeably means you cannot identify either the transaction nor the auction.
 - if you're ok with falling back to a randomly generated value, why not always use a randomly generated value? what's the reason for anything more complicated?
 - `ortb2.ext.tid` should be `ortb2.source.tid` instead (if the above is not a problem).

**Review by dgirardi — APPROVED** [review]

**Comment by patmmccann** [comment]
@nickjacob can you resolve conflicts?

**Comment by nickjacob** [comment]
Sorry about the delay -- @patmmccann fixed merge conflict

**Pull request merged**